plugins/Annotations/API.php

```diff
@@ -296,30 +309,30 @@ private function checkUserCanModifyOrDelete($idSite, $annotation)
      * @throws Exception if the current user is anonymous or does not have view access
      *                   for site w/ id=$idSite.
      */
-    private static function checkUserCanAddNotesFor($idSite)
+    private static function checkUserCanAddNotesFor($idSite): void
     {
         if (!AnnotationList::canUserAddNotesFor($idSite)) {
             throw new Exception("The current user is not allowed to add notes for site #$idSite.");
         }
     }
 
     /**
-     * Utility function, makes sure idSite string has only one site ID and throws if
-     * otherwise.
+     * Throws an exception if the given $idSite does not exist.
+     *
+     * @param int $idSite
+     * @return void
+     * @throws \Piwik\Exception\UnexpectedWebsiteFoundException
      */
-    private function checkSingleIdSite($idSite, $extraMessage)
+    private function checkSiteExists(int $idSite): void
     {
-        // can only add a note to one site
-        if (!is_numeric($idSite)) {
-            throw new Exception("Invalid idSite: '$idSite'. $extraMessage");
-        }
+        new Site($idSite);
```

Comment:
This is a nice idea how to check if a site exists but it might be confusing in a way that the product of it is not used and only the exception is relevant, if thrown.

I guess it's ok as if a site exists, after the first call it's cached, so should be fine from the performance perspective.

Reply:
Yep, It's done that way in a lot of places 🙈